### PR TITLE
Move URI helper methods to utils.uri module

### PR DESCRIPTION
**Pull request by sueann**
## What changes are proposed in this pull request?
 
Consolidate these methods that aren't just used by tracking to `utils.uri`.
 
## How is this patch tested?
 
Unit tests
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
Helper methods for dealing with URIs have been consolidated into the `utils.uri` module. These are internal methods (undocumented) but the change could break existing user code.
 
### What component(s) does this PR affect?
 
- [X] API 
- [X] Python

### How should the PR be classified in the release notes? Choose one:
 
- [X] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section

### CODE REVIEW

File: mlflow/projects/databricks.py [hunk 1]
```diff
@@ -40,8 +40,8 @@ def before_run_validations(tracking_uri, backend_config):
     if backend_config is None:
         raise ExecutionException("Backend spec must be provided when launching MLflow project "
                                  "runs on Databricks.")
-    if not tracking.utils._is_databricks_uri(tracking_uri) and \
-            not tracking.utils._is_http_uri(tracking_uri):
+    if not utils.uri._is_databricks_uri(tracking_uri) and \
```
Comment (mparkhe) [inline mlflow/projects/databricks.py]: some of these functions are used outside the direct package. Does it make sense for use to make them public?

File: mlflow/utils/uri.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+from mlflow.store.db.db_types import DATABASE_ENGINES
+from mlflow.utils import _INVALID_DB_URI_MSG, _validate_db_type_string
+
+
+def _is_local_uri(uri):
```
Comment (mparkhe) [inline mlflow/utils/uri.py]: As noted above -- make them non-private?

**Review by mparkhe — APPROVED** [review]
Great cleanup! One **non-blocking** suggestion to clean up function names ... but might as well cleanup while at it.

**Comment by mparkhe** [comment]
@sueann -- this is a breaking change because the `uri` methods could be external to `mlflow`? But them they aren't documented ones...

**Comment by sueann** [comment]
Since the breaking change label is for release notes, I thought it'd be good to note the changes in the release notes even though it's not _technically_ breaking, since it could break a user's existing code.

**Pull request merged**